File: RegisterInformationApp/app/src/main/java/com/puc/tuangieagenda/viewmodel/DbViewModel.java
package com.puc.tuangieagenda.viewmodel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.puc.tuangieagenda.database.Contatodb;
import com.puc.tuangieagenda.model.Contato;

import java.util.List;

public class DbViewModel extends ViewModel {

    private Contatodb mContatodb;
    public MutableLiveData<List<Contato>> list = new MutableLiveData<>();

    public DbViewModel(Contatodb contatodb) {
        this.mContatodb = contatodb;
    }

    public void insertContact( ) {
        mContatodb.addContact(new Contato("1", "1", "tuannl.com", "-13", "359840"));
        mContatodb.addContact(new Contato("2", "2", "te@email.com", "123.43", "59"));
        mContatodb.addContact(new Contato("3", "3", "tuanne@emaim", "129-13", "359859"));

        refreshList();
    }

    private void refreshList() {
        list.postValue(mContatodb.getAll());
    }

    public MutableLiveData<List<Contato>> getList() {
        return list;
    }
}

File: RegisterInformationApp/app/src/main/java/com/puc/tuangieagenda/viewmodel/DbViewModel.kt
package com.puc.tuangieagenda.viewmodel;

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.puc.tuangieagenda.database.Contatodb
import com.puc.tuangieagenda.model.Contato

class DbViewModele(contatodb: Contatodb) : ViewModel() {

    private val mContatodb = contatodb
    var list: MutableLiveData<List<Contato>> = MutableLiveData()

    fun insertContact() {
        mContatodb.addContact(
            Contato(
                "Tuanne",
                "13",
                "tuanne@email.com",
                "123.456.789-13",
                "35984047259"
            )
        )
        refreshList()
    }

    private fun refreshList() {
        mContatodb.all.let {
            list.postValue(it)
        }
    }
}
